Cython/Compiler/ExprNodes.py

```diff
@@ -5991,8 +5991,14 @@ def nogil_check(self, env):
         func_type = self.function_type()
         if func_type.is_pyobject:
             self.gil_error()
-        elif not func_type.is_error and not getattr(func_type, 'nogil', False):
-            self.gil_error()
+        elif not func_type.is_error:
+            if hasattr(func_type, 'nogil'):
+                if isinstance(func_type.nogil, ExprNode) and not func_type.nogil.value:
```

Comment:
I don't think every `ExprNode` has a `.value`. Isn't this already done differently elsewhere?